src/ui/public/chrome/directives/app_switcher/app_switcher.js

```diff
@@ -1,10 +1,18 @@
 import DomLocationProvider from 'ui/dom_location';
 import { parse } from 'url';
-import { bindKey } from 'lodash';
+import { bindKey, filter } from 'lodash';
 import '../app_switcher/app_switcher.less';
 import uiModules from 'ui/modules';
 import appSwitcherTemplate from './app_switcher.html';
 
+function isNavLinkEnabled(link) {
+  return link.enable === undefined || link.enable;
+}
+
+function isNavLinkShown(link) {
+  return link.show === undefined || link.show;
```

Comment:
@epixa I agree about `id` being a requirement but I think these two properties are different, in that plugins shouldn't be required to provide them. I think its a fair assumption to say that if a plugin is registering a nav item they want it to be shown and enabled by default. So I think they should only need to specify these properties if they want to override them.

I chatted with @cjcenizal out of band about this. I like his suggestion [above](https://github.com/elastic/kibana/pull/7601#discussion_r69335559) because that reflects better the natural usage: the nav link is shown and enabled by default; if a plugin wants to change that, it can set its `hidden` and `disabled` properties.

Thoughts, @epixa?